test/e2e/azure_machinepool_drain.go

```diff
@@ -157,13 +165,39 @@ func testMachinePoolCordonAndDrain(ctx context.Context, mgmtClusterProxy, worklo
 			return err
 		}
 
-		decreasedReplicas := *owningMachinePool.Spec.Replicas - int32(1)
+		decreasedReplicas = *owningMachinePool.Spec.Replicas - int32(1)
 		owningMachinePool.Spec.Replicas = &decreasedReplicas
 		return helper.Patch(ctx, owningMachinePool)
 	}, 3*time.Minute, 3*time.Second).Should(Succeed())
 
-	// TODO setup a watcher to validate expected 2nd order drain outcomes
-	// https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2159
+	By(fmt.Sprintf("checking for a machine to start draining for machine pool: %s/%s", amp.Namespace, amp.Name))
+	Eventually(func() error {
+		ampmls, err := getAzureMachinePoolMachines(ctx, mgmtClusterProxy, workloadClusterProxy, amp)
+		if err != nil {
+			LogWarning(errors.Wrap(err, "failed to list the azure machine pool machines").Error())
+			return err
+		}
+
+		for i := range ampmls {
+			if conditions.Has(&ampmls[i], clusterv1.DrainingSucceededCondition) && conditions.IsFalse(&ampmls[i], clusterv1.DrainingSucceededCondition) {
```

Comment:
nit: I think `Has` is redundant here since `IsFalse` returns false if the condition doesn't exist:
```suggestion
			if conditions.IsFalse(&ampmls[i], clusterv1.DrainingSucceededCondition) {
```